flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -295,31 +295,42 @@ public DataTypeFactory getDataTypeFactory() {
      *
      * @param catalogName the given catalog name under which to create the given catalog
      * @param catalogDescriptor catalog descriptor for creating catalog
+     * @param ignoreIfExists if false exception will be thrown if a catalog exists.
      * @throws CatalogException If the catalog already exists in the catalog store or initialized
      *     catalogs, or if an error occurs while creating the catalog or storing the {@link
      *     CatalogDescriptor}
      */
-    public void createCatalog(String catalogName, CatalogDescriptor catalogDescriptor)
+    public void createCatalog(
+            String catalogName, CatalogDescriptor catalogDescriptor, boolean ignoreIfExists)
             throws CatalogException {
         checkArgument(
                 !StringUtils.isNullOrWhitespaceOnly(catalogName),
                 "Catalog name cannot be null or empty.");
         checkNotNull(catalogDescriptor, "Catalog descriptor cannot be null");
 
         if (catalogStoreHolder.catalogStore().contains(catalogName)) {
-            throw new CatalogException(
-                    format("Catalog %s already exists in catalog store.", catalogName));
+            if (!ignoreIfExists) {
```

Comment:
The logic is a little bit cumbersome, what about

```java
public void createCatalog(
        String catalogName, CatalogDescriptor catalogDescriptor, boolean ignoreIfExists)
        throws CatalogException {
    checkArgument(
            !StringUtils.isNullOrWhitespaceOnly(catalogName),
            "Catalog name cannot be null or empty.");
    checkNotNull(catalogDescriptor, "Catalog descriptor cannot be null");

    boolean catalogExistsInStore = catalogStoreHolder.catalogStore().contains(catalogName);
    boolean catalogExistsInMemory = catalogs.containsKey(catalogName);

    if (catalogExistsInStore || catalogExistsInMemory) {
        if (!ignoreIfExists) {
            throw new CatalogException(
                    format("Catalog %s already exists.", catalogName));
        }
    } else {
        // Store the catalog in the catalog store
        catalogStoreHolder.catalogStore().storeCatalog(catalogName, catalogDescriptor);
        
        // Initialize and store the catalog in memory
        Catalog catalog = initCatalog(catalogName, catalogDescriptor);
        catalog.open();
        catalogs.put(catalogName, catalog);
    }
}

```

Reply:
good idea